testing_local_changes.md

```diff
@@ -0,0 +1,115 @@
+which is key component used by the dataplane-operator that deploys EDPM
+nodes. The dataplane-operator's CRD includes support for specifying additional
+volume mounts for the ansibleee pods, which provides a mechanism for accessing
+a local copy of edpm-ansible. This makes it possible to develop and test local
+changes edpm-ansible without having to build and deploy a new
+openstack-ansibleee-runner container image.
+
+# Provide NFS access to your edpm-ansible directory
+
+The technique described here uses NFS to access the edpm-ansible directory on
+your development system, so you'll need to install an NFS server and create
+an appropriate export on your development system. Of course, this implies
+your OpenShift deployment that runs the dataplane-operator has access to
+the NFS server, including any required firewall rules.
+
+* [EL 8 instructions](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/exporting-nfs-shares_deploying-different-types-of-servers#assembly_configuring-the-nfs-server-to-run-behind-a-firewall_exporting-nfs-shares)
+* [EL 9 instructions](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/managing_file_systems/exporting-nfs-shares_managing-file-systems#assembly_configuring-the-nfs-server-to-run-behind-a-firewall_exporting-nfs-shares)
+
+When using OpenShift Local (aka CRC), your export will be something like this:
+
+```
+% cat &lt;&lt;EOF &gt;/etc/exports
+${HOME}/edpm-ansible 192.168.130.0/24(rw,sync,no_root_squash)
+EOF
+
+% exportfs -r
+```
+
+**NOTE:** CRC installs its own firewall rules, which likely will need to be
+adjusted depending on the location of your NFS server. If your edpm-ansible
+directory is on the same system that hosts your CRC, then the simplest thing
+to do is insert a rule that essentially circumvents the other rules:
+
+```
+% nft add rule inet firewalld filter_IN_libvirt_pre accept
+```
+
+# Create edpm-ansible PV and PVC
+
+Create an NFS PV, and a PVC that can be mounted on the ansibleee pods.
+
+**NOTE:** While it's possible to add an NFS volume directly to a pod, the
+default k8s Security Context Constraint (SCC) for non-privileged pods does
+not permit NFS volume mounts. The approach of using an NFS PV and PVC works
+just as well, and avoids the need to fiddle with SCC policies.
+
+```
+% # E.g. ${HOME}/edpm-ansible
+% NFS_SHARE=&lt;Path to your edpm-ansible directory&gt;
+% NFS_SERVER=&lt;IP of your NFS server&gt
+% cat &lt;&lt;EOF &gt;edpm-ansible-storage.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: edpm-ansible
+spec:
+  capacity:
+    storage: 1Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadOnlyMany
+  # IMPORTANT! The persistentVolumeReclaimPolicy must be "Retain" or else
+  # your code will be deleted when the volume is reclaimed!
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: edpm-ansible
+  mountOptions:
```

Comment:
when we dicussed this breifly on the call you also confirmed that
while this example is using nfs you could in theory use cephfs or any other volume driver that could fullfile the access mode in this case ReadOnlyMany (ROX)

NFS is definally on of the simpler solution to meet that crtia 

in terms of the mount option you could is - hard actually required
hard mounts could cause block io issue in the oepnshif cluster if the nfs connection was to drop but for development that probably ok.

Reply:
NFS was the simplest, logical choice for me because I already had the NFS server infrastructure in place (for testing an NFS cinder backend). I'm sure there are other possibilities.

Yeah, I think the "hard" mount option was just part of another example I looked at when I started playing with this, and agree it's not something you'd want to do in a production environment.

Reply:
ack